src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/Templates/MauiAndroid/MainApplication.cs

```diff
@@ -2,15 +2,18 @@
 using Android.App;
 using Android.Runtime;
 using Microsoft.Maui;
+using Microsoft.Maui.Hosting;
 
 namespace MauiAndroidApp1
 {
 	[Application]
-	public class MainApplication : MauiApplication<Startup>
+	public class MainApplication : MauiApplication
 	{
 		public MainApplication(IntPtr handle, JniHandleOwnership ownership)
 			: base(handle, ownership)
 		{
 		}
-	}
-}
\ No newline at end of file
+
+        protected override MauiApp CreateMauiApp() => MauiProgram.CreateMauiApp();
```

Comment:
It seems like we have to always add this boilerplate; I couldn't dive in to figure out why this doesn't just exist by default in `MauiApplication` but feels like it should.

Reply:
this is a template file we add to a .NET MAUI Android project. The template file comes from the .NET MAUI default template. I believe this comment is more towards the .NET MAUI implementation and I would suggest adding this issue in the [.NET MAUI Github](https://github.com/dotnet/maui). The purpose of this extension is to help with the migration process from Xamarin.Forms project to .NET MAUI project and this template file is newly required in a .NET MAUI project and as mentioned before, it comes from the default .NET MAUI template. 

More information about .NET MAUI can also be found in the documentation [here](https://docs.microsoft.com/en-us/dotnet/maui/). Details about Migration and steps for migration can be found [here](https://github.com/dotnet/maui/wiki/Migrating-from-Xamarin.Forms-(Preview)).